src/libs/utilsTs.ts

```diff
@@ -18,3 +18,25 @@ export const urlEncodeParams = (params: Record<string, string | number | boolean
  */
 export const getCloudPlatform = (dataset: DatasetModel): CloudPlatform | undefined =>
   _.first(dataset.storage?.map((s) => s.cloudPlatform));
+
+/**
+ * @param id {string} the id of the snapshot access request
+ * @param snapshotName {string} the user specified name of the snapshot access request
+ *
+ * @return {string} An underscore joined name and id with all dashes and spaces converted to
+ * underscores and all non-alphanumeric or underscore character stripped out.
+ * It also trims all leading underscores
+ */
+export const generateSnapshotNameFromAccessRequestInformation = (
```

Comment:
instead of handling on the front end - should we handle it on the back end? because if we just handle it here it will still be a problem for API users. I made a ticket for this recently: https://broadworkbench.atlassian.net/browse/DC-1128

Reply:
I think API users could use whatever name they want, and may prefer to get an exception then have their name they provide be changed out from under them.

Reply:
Okay!